### PR TITLE
add a new header which combines LinearOperator and PackagedOperation

### PR DESCRIPTION
**Pull request by davydden**
fixes https://github.com/dealii/dealii/issues/2504

### CODE REVIEW

**Comment by luca-heltai** [comment]
I'd include also the block variants.

**Comment by davydden** [comment]
i also would like to adjust the group description to mention the new header, but did not find yet where is it located.

**Comment by luca-heltai** [comment]
This is fine for me. Maybe let's wait for @bangerth, who was opposed to this solution in the beginning. :)

**Comment by bangerth** [comment]
I'll let you make the call -- I see a lot of excitement about these linear operators, and while I usually err on the side of "reduce the number of surprises" (in this case when using `auto` for certain operations), I also don't want to stand in the way of progress.

So do as you think is appropriate!

File: include/deal.II/lac/linear_operator_tools.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// many usage cases lead to a combination of LinearOperator and
+// PackagedOperation. To ease the pain of reading compilation errors, just include
+// all headers we ever need to use LO and friends in one place:
+#include <deal.II/lac/linear_operator.h>
+#include <deal.II/lac/block_linear_operator.h>
+#include <deal.II/lac/packaged_operation.h>
+#include <deal.II/lac/constrained_linear_operator.h>
+
+// keep Schur complement for completeness
+#include <deal.II/lac/schur_complement.h>
```
Comment (tamiko) [inline include/deal.II/lac/linear_operator_tools.h]: Why? :-)

Comment (tamiko) [inline include/deal.II/lac/linear_operator_tools.h]: I.e. what is the rationale behind also including ```schur_complement.h```?

Comment (davydden) [inline include/deal.II/lac/linear_operator_tools.h]: to keep everything together. Makes life easier and does not hurt. Also consistent with naming: ``linear_operator_tools.h`, that is -- everything that is implemented using LO.

Comment (tamiko) [inline include/deal.II/lac/linear_operator_tools.h]: Oh, wow. I totally forgot that ```schur_complement.h``` is implemented in terms of ```LinearOperator```. Can you just strip the comment and group it with the ```constrained_linear_operator.h``` include?

**Comment by davydden** [comment]
@tamiko ping.

**Comment by davydden** [comment]
@tamiko any comments from you as co-author of LO classes?

**Comment by Rombur** [comment]
I think we have let enough time for everybody to give their opinion on this PR. Since nobody opposes it, I am going to merge it.

**Pull request merged**